packages/components/src/public/QueryModel/CustomizeGridViewModal.tsx

```diff
@@ -113,6 +138,31 @@ export const CustomizeGridViewModal: FC<Props> = memo(props => {
         return columnsInView.findIndex(col => col === column) !== -1;
     }
 
+    const onDropField = useCallback((dropResult: DropResult): void => {
+        const { destination, draggableId, source } = dropResult;
+        if (destination === null || source.index === destination.index) {
+            return;
+        }
+        const { index } = destination;
+
+        const colInMotion = columnsInView[source.index];
+        let updatedColumns = columnsInView.filter(col => col.index != draggableId);
+        updatedColumns = [...updatedColumns.slice(0, index), colInMotion, ...updatedColumns.slice(index) ];
+        setColumnsInView(updatedColumns);
+        if (source.index === selectedIndex) {
+            setSelectedIndex(index);
```

Comment:
as mentioned in test plan, also need to handle the case where there was a selected index and something else was dragged from above it to below it.